File: src/main/java/com/cocktailsguru/app/cocktail/domain/ingredient/BaseIngredient.java
package com.cocktailsguru.app.cocktail.domain.ingredient;

import lombok.Data;

import javax.persistence.*;

@MappedSuperclass
@Data
abstract class BaseIngredient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String nameGrouped;

    @Column(name = "desc")
    private String description;

    @Column(name = "imgFileName")
    private String imageName;

    private Integer numShowed;

    @OneToOne
    @JoinColumn(name = "categoryFK")
    private IngredientType ingredientType;


}

File: src/main/java/com/cocktailsguru/app/ingredient/domain/BaseIngredient.kt
package com.cocktailsguru.app.ingredient.domain

import javax.persistence.*

@MappedSuperclass
abstract class BaseIngredient(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        open val id: Long,
        open var name: String,
        @Column(name = "nameGrouped")
        open var nameGrouped: String,
        @Column(name = "desc")
        open var description: String,
        @Column(name = "imgFileName")
        open var imageName: String,
        @Column(name = "numShowed")
        open var numShowed: Int,
        @OneToOne
        @JoinColumn(name = "categoryFK")
        open var ingredientType: IngredientType
)